doc/design/manifest.md

```diff
@@ -0,0 +1,585 @@
+
+#### one component
+
+e.g., `tiup update foo`
+
+* Download and verify timestamp.json, if the hash of snapshot.json is unchanged, then finish with nothing to update.
+  - If timestamp.json cannot be verified, it is possible that the key has changed. Retry after updating root.json
+* Update root.json
+* Download and verify timestamp.json (finish if it has not changed)
+* Use timestamp.json to download and verify snapshot.json
+* Use snapshot.json to download and verify index.json
+* Check index.json for new default components, install them using the install/update component flow
+* If the component has been yanked, inform the user.
+* Use index.json to download and verify the component manifest (finish if the signature has not changed)
+* For each platform
+  - for each major version the user has installed, find the most recent, non-yanked recent version.
+  - if the most recent version is more recent than the installed version, download and verify it, then install the new component and delete the old component.
+  - if there is a new major version, download the most recent version of it and tell the user
+
+Note that we do not specify what should happen to existing versions or what should be communicated to the user
+
+#### all components
+
+e.g., `tiup update --all`
+
+* Download and verify timestamp.json, if the hash of snapshot.json is unchanged, then finish with nothing to update.
+  - If timestamp.json cannot be verified, it is possible that the key has changed. Retry after updating root.json
+* Update root.json
+* Download and verify timestamp.json (finish if it has not changed)
+* Use timestamp.json to download and verify snapshot.json
+* Use snapshot.json to download and verify index.json
+* Check index.json for new default components, install them using the install/update component flow
+* For each component the user has installed (excluding TiUp)
+  - If the component has been yanked, inform the user and continue to next component.
+  - Use index.json to download and verify the component manifest (continue to next component if the signature has not changed)
+  - For each platform
+    * for each major version the user has installed, find the most recent, non-yanked version.
+    * if the most recent version is more recent than the installed version, download and verify it, then install the new component and delete the old component.
+    * if there is a new major version, download the most recent version of it and tell the user
+
+#### Self update
+
+e.g., `tiup update --self`
+
+Follow steps for one component workflow.
+
+### Add a new owner
+
+Probably an implicit action when a user publishes their first component.
+
+* Generate a new key pair in the client. The private key is stored securely.
+* Update root.json
+* Send new owner request from TiUp client to server including the client's public key but not private key; request must be sent over TLS to ensure that the public key is not tampered with.
```

Comment:
Where the public key comes from? And where we store it?

Reply:
Line 420 - the client generates it; "key pair" = a public key and a private key. The public key is stored in index.json (the private key is stored by the client)

Reply:
What if the user lost the key (eg. reinstall tiup)? Can he control his old packages?

Reply:
There is a flow for restoring access, I should specify that keys may be backed up